File: fixtures/scale/php/large/src/Families/Adler32/Adler32Runner.php
<?php

namespace CipherLab\Families\Adler32;

use CipherLab\Abstractions\VectorOutcome;
use CipherLab\Engine\CorrectnessEngine;

// Correctness entry point for the adler32 family; wraps CorrectnessEngine
// with this family's cipher and known-answer vectors.
class Adler32Runner
{
    private CorrectnessEngine $engine;

    public function __construct()
    {
        $this->engine = new CorrectnessEngine();
    }

    public function check(): VectorOutcome
    {
        return $this->engine->verify(new Adler32Cipher(), Adler32Vectors::all());
    }
}

File: fixtures/scale/swift/large/Families/Adler32/Adler32Runner.swift
// Correctness entry point for the adler32 family; wraps CorrectnessEngine
// with this family's cipher and known-answer vectors.
struct Adler32Runner {
    private let engine = CorrectnessEngine()

    func check() -> VectorOutcome {
        let cipher = Adler32Cipher()
        let vectors = Adler32Vectors.all()
        return engine.verify(cipher, vectors)
    }
}
